### DISCUSSION TITLE
npx prisma db seed throws ERR_UNKNOWN_FILE_EXTENSION

### DISCUSSION THREAD

**Discussion by danielniccoli** [prisma/prisma#20369] [Q&A] [answered]
How do I get seeding to work? Running `npx prisma db seed` results in 

```
Environment variables loaded from .env
Running seed command `ts-node prisma/seed.ts` ...
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for C:\Users\Daniel\Documents\source\myproject\prisma\seed.ts
    at new NodeError (node:internal/errors:405:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:99:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:142:36)
    at defaultLoad (node:internal/modules/esm/load:86:20)
    at DefaultModuleLoader.load (node:internal/modules/esm/loader:281:26)
    at DefaultModuleLoader.moduleProvider (node:internal/modules/esm/loader:192:22)
    at new ModuleJob (node:internal/modules/esm/module_job:63:26)
    at DefaultModuleLoader.#createModuleJob (node:internal/modules/esm/loader:216:17)
    at DefaultModuleLoader.getJobFromResolveResult (node:internal/modules/esm/loader:169:34)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:154:17) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

An error occurred while running the seed command:
Error: Command failed with exit code 1: ts-node prisma/seed.ts
```

I also tried:

```json
"prisma": {
  "seed": "ts-node --compiler-options {\"module\":\"CommonJS\"} prisma/seed.ts"
},
```

and

```
"prisma": {
  "seed": "ts-node --transpile-only prisma/seed.ts"
},
```

and all three commands with `ts-node-esm` instead of `ts-node`.

This is my tsconfig.json:

```json
{
	"extends": "./.svelte-kit/tsconfig.json",
	"compilerOptions": {
		"allowJs": true,
		"checkJs": true,
		"esModuleInterop": true,
		"forceConsistentCasingInFileNames": true,
		"resolveJsonModule": true,
		"skipLibCheck": true,
		"sourceMap": true,
		"strict": true
	}
	// Path aliases are handled by https://kit.svelte.dev/docs/configuration#alias
	//
	// If you want to overwrite includes/excludes, make sure to copy over the relevant includes/excludes
	// from the referenced tsconfig.json - TypeScript does not merge them in
}
```

and `./.svelte-kit/tsconfig.json`
```json
{
	"compilerOptions": {
		"paths": {
			"$lib": [
				"../src/lib"
			],
			"$lib/*": [
				"../src/lib/*"
			]
		},
		"rootDirs": [
			"..",
			"./types"
		],
		"importsNotUsedAsValues": "error",
		"isolatedModules": true,
		"preserveValueImports": true,
		"lib": [
			"esnext",
			"DOM",
			"DOM.Iterable"
		],
		"moduleResolution": "node",
		"module": "esnext",
		"target": "esnext",
		"ignoreDeprecations": "5.0"
	},
	"include": [
		"ambient.d.ts",
		"./types/**/$types.d.ts",
		"../vite.config.ts",
		"../src/**/*.js",
		"../src/**/*.ts",
		"../src/**/*.svelte",
		"../tests/**/*.js",
		"../tests/**/*.ts",
		"../tests/**/*.svelte"
	],
	"exclude": [
		"../node_modules/**",
		"./[!ambient.d.ts]**",
		"../src/service-worker.js",
		"../src/service-worker.ts",
		"../src/service-worker.d.ts"
	]
}
```

**Comment by nurul3101** [comment]
Hello @danielniccoli 👋 

Does your seed script work if you run this command directly?

```ts
npx ts-node prisma/seed.ts
```

**Comment by danielniccoli** [comment]
Hi @nurul3101, same issue.

**Reply by janpio** [reply]
That then only means that your seed script or Typescript is not configured correctly to directly execute scripts like that. This is more a Typescript or maybe Svelte-Kit (where you got your tsconfig from) question than a Prisma one. Prisma just tries to execute the command you provide to it.

**Reply by danielniccoli** [reply]
Yes, I thought as much, which is why I opened a discussion and not an issue. Do you have a suggestion how to fix this? I'm new to TypeScript and don't know.

**Reply by janpio** [reply]
(See the other comment from my colleague - he actually knows how this stuff works 😆 )

**Comment by millsp** [comment, marked as answer]
Hey @danielniccoli, have you tried `npx ts-node --esm prisma/seed.ts`? Alternatively, I know [tsx](https://github.com/esbuild-kit/tsx) works well for Svelte folks (and it's also much faster), you could also try `tsx prisma/seed.ts`.

**Reply by danielniccoli** [reply]
tsx worked, thanks.

**Reply by peturh** [reply]
Ran:

```
npm install tsx --save-dev
```

Added 
```
"prisma": {
		"seed": "tsx prisma/seed.ts"
	},
```

to package.json

and

ran the migration steps.

```
npx prisma migrate dev --name init
```

**Reply by chaya-tan** [reply]
works like a charm, thanks!

**Reply by cihadaydemir** [reply]
just wanted to give an update, this issue is still a problem and seeding with `tsx prisma/seed.ts` still works 🚀

**Reply by umanda** [reply]
`tsx` is working for me too.

**Reply by ebe25** [reply]
hey but as far as tsx is concerned it does not transpile right ? then what would be the point using ts in dev-server? just curious.

**Reply by ergs22** [reply]
> Ran:
> 
> ```
> npm install tsx --save-dev
> ```
> 
> Added
> 
> ```
> "prisma": {
> 		"seed": "tsx prisma/seed.ts"
> 	},
> ```
> 
> to package.json
> 
> and
> 
> ran the migration steps.
> 
> ```
> npx prisma migrate dev --name init
> ```

THANK U :')

**Reply by philsmithies** [reply]
> > Ran:
> > ```
> > npm install tsx --save-dev
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Added
> > ```
> > "prisma": {
> > 		"seed": "tsx prisma/seed.ts"
> > 	},
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to package.json
> > and
> > ran the migration steps.
> > ```
> > npx prisma migrate dev --name init
> > ```
> 
> THANK U :')

thanks for the detailed steps!

**Reply by dhanush-tnva** [reply]
update: this issue is still a problem and seeding with tsx prisma/seed.ts still works 
ps: using this in my remix project

**Comment by kirso** [comment]
None of the solutions above worked for me, but this did:

Install tsx
`npm install tsx`

In package.json
```
{
  "scripts": {
    "db-seed": "NODE_ENV=development prisma db seed"
  },
  "prisma": {
    "seed": "tsx prisma/seed.ts"
  }
}
```
In prisma/seed.ts

```
import { prisma } from "../src/server/db";

async function main() {
  const id = "cl9ebqhxk00003b600tymydho";
  await prisma.example.upsert({
    where: {
      id,
    },
    create: {
      id,
    },
    update: {},
  });
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });
```

Then run

`npm run db-seed`

**Comment by AdelForplikter** [comment]
I was following instructions [here](https://www.prisma.io/blog/sveltekit-prisma-kvCOEoeQlC) from [prisma.io](https://www.prisma.io): 
Rename seed.ts to seed.js and 
`npx ts-node prisma/seed.js`
also worked here.

**Comment by likui628** [comment]
`node --loader ts-node/esm prisma/seed.ts`

**Comment by RobinHoodCoder** [comment]
Pretty annoying to have to read through all this, just to get the Prisma tutorial working in my scratch project. Using Vite+React out of the box here. Maybe nice to make it work out of the box with this popular stack? Easiest will be to just change the tutorial a bit.

**Comment by xjdrew** [comment]
correct answer is here: https://stackoverflow.com/questions/69580704/nextjs-run-a-typescript-script-on-the-server
